miner/src/work_notify.rs

```diff
@@ -91,51 +83,27 @@ impl NotifyWork for WorkPoster {
 			r#"{{ "result": ["0x{:x}","0x{:x}","0x{:x}","0x{:x}"] }}"#,
 			pow_hash, seed_hash, target, number
 		);
-		let mut client = self.client.lock();
-		for u in &self.urls {
-			if let Err(e) = client.request(u.clone(), PostHandler { body: body.clone() }) {
-				warn!("Error sending HTTP notification to {} : {}, retrying", u, e);
-				// TODO: remove this once https://github.com/hyperium/hyper/issues/848 is fixed
-				*client = WorkPoster::create_client();
-				if let Err(e) = client.request(u.clone(), PostHandler { body: body.clone() }) {
-					warn!("Error sending HTTP notification to {} : {}", u, e);
-				}
+		let urls = self.urls.clone();
+
+		self.remote.spawn_fn(move |handle| {
+			let client = Config::default()
+				.keep_alive(true).build(handle);
+
+			let mut future: Box<Future<Item=(), Error=()>> = Box::new(future::ok(()));
+			for url in urls {
+				let mut req = Request::new(Method::Post, url.clone());
```

Comment:
Wouldn't it be better to use `fetch` instead of raw `hyper`?

Reply:
I'm not sure whether we can use `fetch` to send POST request. The `fetch` crates looks like to use a background thread and is hard coded to only send GET request? https://github.com/paritytech/parity/blob/master/util/fetch/src/client.rs#L216

Reply:
this is changed in #6422 which will be merged in a second https://github.com/paritytech/parity/pull/6422/files#diff-3eef53ff37b934cea2e67f6f251a2377R131

Reply:
Got it. I'll try to get a version based on `fetch` once that PR is merged.